File: CrawlHTMLLab/static/enums/constant.js
//import { ResearchPaperRadio, ResearchPaperInput } from "./taskEnums/researchPaper.js";

let ResearchPaperInput = {}
let ResearchPaperRadio = {}

export const CrawlingOptions = {
    RESEARCH_PAPER: {
        value: 1,
        label: 'Research Paper',
        placeholder: 'Enter your keyword',
        inputLabel: 'Keyword',
        fields: {
            radio: ResearchPaperRadio,
            input: ResearchPaperInput
        },
        name: "",
        area: ".research-paper-area"
    },
    NEWS: {
        value: 2,
        label: 'News',
        placeholder: 'Enter keyword',
        inputLabel: 'Keyword',
        name: "",
        area: ".news-area"
    },
    FACEBOOK: {
        value: 3,
        label: 'Facebook',
        placeholder: 'Enter username',
        inputLabel: 'Username',
        name: "",
        area: ".facebook-area"
    },
    GOOGLE_IMAGE: {
        value: 4,
        label: 'Google Image',
        placeholder: 'Enter keyword',
        inputLabel: 'Keyword',
        name: "",
        area: ".google-image-area"
    }
}

File: CrawlHTMLLab/static/enums/constant.py
from taskEnums import facebook, news, researchPaper, googleImage

CRAWLING_OPTIONS = {
    'RESEARCH_PAPER': {
        'value': 1,
        'label': 'Research Paper',
        'fields': {
            "radio": researchPaper.REASEARCH_PAPER_RADIO,
            "input": researchPaper.RESEARCH_PAPER_INPUT
        }
    },
    'NEWS': {
        'value': 2,
        'label': 'News',
        "fields": {
            "input": news.NEWS_INPUT
        }
    },
    'FACEBOOK': {
        'value': 3,
        'label': 'Facebook',
        "fields": {
            "input": facebook.FACEBOOK_INPUT
        }
    },
    'GOOGLE_IMAGE': {
        'value': 4,
        'label': 'Google Image',
        "fields": {
            "input": googleImage.GOOGLE_IMAGE_INPUT
        }
    }
}

SELECT_VALUE = {
  1: CRAWLING_OPTIONS['RESEARCH_PAPER'],
  2: CRAWLING_OPTIONS['NEWS'],
  3: CRAWLING_OPTIONS['FACEBOOK'],
  4: CRAWLING_OPTIONS['GOOGLE_IMAGE']
}